File: disappeared_element.py
# Time Complexity: O(n)
# Space Complexity: O(1)
class Solution(object):
    def findDisappearedNumbers(self, nums):
        #         Defining output array
        out_arr = []

        #         Iterate over array and take the number  - 1 index and make that value at that index as negative
        for i in range(len(nums)):
            curr_index = abs(nums[i]) - 1
            if nums[curr_index] > 0:
                nums[curr_index] = nums[curr_index] * -1

        #         Again iterate over input array and restore array to its original state and result will be positive elements index + 1
        for i in range(len(nums)):
            if nums[i] > 0:
                out_arr.append(i + 1)
            else:
                nums[i] = nums[i] * -1
        #         return output array
        return out_arr

        """
        :type nums: List[int]
        :rtype: List[int]
        """

File: disappeared-element.cpp
/*The code utilizes the fact that each element in the array represents a number, and we can use the array itself to mark the presence of numbers.
It first iterates through the array and negates the value at the index corresponding to the absolute value of each element encountered.
After negating, the indices with positive values indicate the missing numbers, and their corresponding numbers are added to the result vector.
Time Complexity: O(n), where n is the size of the input array.
Space Complexity: O(1) as the result vector is not considered in the space complexity calculation. The algorithm modifies the input array in-place.*/
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
     vector<int> findDisappearedNumbers(vector<int>& nums) {
       vector<int>result;
       int n=nums.size();
       for(int i=0;i<n;i++){
           int number=abs(nums[i]);
           int idx=number-1;
           if(nums[idx]>0){
               nums[idx]*=-1;
           }
       }
       for(int i=0;i<n;i++){
           if(nums[i]>0){
               result.push_back(i+1);
           }
       }
       return result;
    }
};